osu.Game/Screens/Play/SubmittingPlayer.cs

```diff
@@ -234,6 +233,16 @@ protected override void StartGameplay()
             spectatorClient.BeginPlaying(token, GameplayState, Score);
         }
 
+        protected override void ConcludeGameplay()
+        {
+            base.ConcludeGameplay();
+
+            spectatorClient.EndPlaying(GameplayState);
+
+            // compare: https://github.com/ppy/osu/blob/ccf1acce56798497edfaf92d3ece933469edcf0a/osu.Game/Screens/Play/Player.cs#L848-L851
+            submitScore(Score.DeepClone()).FireAndForget();
```

Comment:
This previously ran the other way around, with the `submitScore` being awaited before sending the `EndPlaying` request.

I'm not sure whether this was an actual ordering requirement or not. @bdach may have a better idea..

Reply:
Without reading too hard into this yet (will probably do later), the ordering here was important for https://github.com/ppy/osu/pull/21686.

Reply:
Before I do anything here I would want a 3-way confirmation that the correct fix here is a server-side one.

Reply:
To clarify, the path forward would be to remove the pubsub completely and convert to a polling loop?

Reply:
As far as I'm aware, yes, that is the proposition.

I'm not overly attached to the pubsub. My only worry is whether polling is going to perform well enough to hold up performance-wise. I am not sure how to tell if it will in any way other than just writing that change and deploying it live.